js-envs/src/main/scala/org/scalajs/jsenv/phantomjs/PhantomJSEnv.scala

```diff
@@ -241,13 +243,15 @@ class PhantomJSEnv(
       }
     }
 
-    def receive(): String = synchronized {
+    def receive(timeout: Duration): String = synchronized {
       if (recvBuf.isEmpty && !awaitConnection())
         throw new ComJSEnv.ComClosedException("Phantom.js isn't connected")
 
+      val deadline = OptDeadline(timeout)
+
       @tailrec
       def loop(acc: String): String = {
-        val frag = receiveFrag()
+        val frag = receiveFrag(deadline)
```

Comment:
This is not consistent with how the NodeJS channel handles timeouts (it looses previous fragments, if a timeout happens). You need to make a decision here:

1. If a timeout occurs, the channel is bad, nothing can be done anymore
2. If a timeout occurs, the next call to receive will give the message the call before has timeout out.